qiskit/transpiler/passes/synthesis/plugin.py

```diff
@@ -243,6 +246,36 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
 :func:`qiskit.transpiler.passes.synthesis.plugin.unitary_synthesis_plugin_names`
 function.
 
+High Level Synthesis Plugins
+----------------------------
+
+To use a high level synthesis you leverage the :class:`.HLSConfig` class to
+create an input configuration for how to synthesize multiple high level
+plugins. For example::
+
+    HLSConfig(permutation=["acg"], clifford=["layers"], linear_function=["pmh"])
+
+will create a high level synthesis configuration that will use the ``acg`` plugin
+for synthesizing :class:`.PermutationGate` objects, the ``layers`` plugin
+for synthesizing :class:`.Clifford` objects, and the ``pmh`` plugin for synthesizing
+:class:`.LinearFunction` objects. The :class:`.HLSConfig` class
+has the flexibility to specify multiple plugins in sequence which will enable
+trying multiple synthesis plugins and using the first one that successsfully
+synthesizes the object and also to pass tuples of the plugin along with options
+to adjust the behavior of the synthesis plugin.
```

Comment:
```suggestion
creates a high-level synthesis configuration that uses the ``acg`` plugin
for synthesizing :class:`.PermutationGate` objects, the ``layers`` plugin
for synthesizing :class:`.Clifford` objects, and the ``pmh`` plugin for synthesizing
:class:`.LinearFunction` objects.
For each high-level object, the list of given plugins are tried in sequence until one of them succeeds (in the example above, each list only contains a single plugin). In addition to specifying a plugin by its name, one can also instead pass a ``(name, options)`` tuple, where the second element of the tuple contains options for the plugin.
```

@mtreinish I suggested a significant rewording. Is it correct? Also, what is the format of the options for the plugin?

Reply:
Yeah, this looks good.

 `options` is a dictionary where the specifics are defined by a given plugin. Implementation wise they're the kwargs dict that gets passed to the plugins `.run()` method. But, for the purposes of an `HLSConfig` it's just a dictionary with arbitrary keys and values.